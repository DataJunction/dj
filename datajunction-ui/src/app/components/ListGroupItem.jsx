import { Component } from 'react';

export default class ListGroupItem extends Component {
  render() {
    const { label, value } = this.props;
    return (
      <div className="list-group-item d-flex">
        <div className="d-flex gap-2 w-100 justify-content-between py-3">
          <div>
            <h6 className="mb-0 w-100">{label}</h6>
            <p className="mb-0 opacity-75">{value}</p>
          </div>
        </div>
      </div>
    );
  }
}
