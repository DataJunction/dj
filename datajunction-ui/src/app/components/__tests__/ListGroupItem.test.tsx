import * as React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';

import ListGroupItem from '../ListGroupItem';

const renderer = createRenderer();

describe('<ListGroupItem />', () => {
  it('should render and match the snapshot', () => {
    renderer.render(
      <ListGroupItem label="Name" value={<span>Something</span>} />,
    );
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
