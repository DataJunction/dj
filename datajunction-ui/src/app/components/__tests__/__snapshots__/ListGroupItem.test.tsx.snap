// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListGroupItem /> should render and match the snapshot 1`] = `
<div
  className="list-group-item d-flex"
>
  <div
    className="d-flex gap-2 w-100 justify-content-between py-3"
  >
    <div>
      <h6
        className="mb-0 w-100"
      >
        Name
      </h6>
      <p
        className="mb-0 opacity-75"
      >
        <span>
          Something
        </span>
      </p>
    </div>
  </div>
</div>
`;
